packages/babel-generator/src/buffer.js

```diff
@@ -38,10 +38,25 @@ export default class Buffer {
   get(): Object {
     this._flush();
 
-    return {
+    const map = this._map;
+    const result = {
       code: trimEnd(this._buf.join("")),
-      map: this._map ? this._map.get() : null,
+      map: null,
+      rawMappings: map && map.getRawMappings(),
     };
+
+    if (map) {
+      Object.defineProperty(result, "map", {
+        get() {
+          return this.map = map.get();
```

Comment:
Oops, I unintentionally removed `configurable: true, enumerable: true` while cleaning this up a little. Good catch.

The getter uses `=` to reuse the setter below. The setter exists to make the `.map` maintain the current behavior of the property. I.e. the whole object continues to work as it used to do, but the source map is created lazily.